cluster/manifests/wiz/sensor/dameonset.yaml

```diff
@@ -0,0 +1,223 @@
+    type: RollingUpdate
+
+  template:
+    metadata:
+      labels:
+        helm.sh/chart: wiz-sensor-1.0.4760
+        image/tag: v1
+        app.kubernetes.io/name: wiz-sensor
+        app.kubernetes.io/instance: wiz-sensor
+        app.kubernetes.io/version: "1.0.4708"
+        app.kubernetes.io/managed-by: Helm
+      annotations:
+        container.apparmor.security.beta.kubernetes.io/wiz-sensor: unconfined
+        cluster-autoscaler.kubernetes.io/enable-ds-eviction: "true"
+        node-ready.cluster.zalando.org/exclude: "true"
+
+    spec:
+      serviceAccountName: wiz-sensor
+      affinity:
+        nodeAffinity:
+            requiredDuringSchedulingIgnoredDuringExecution:
+              nodeSelectorTerms:
+              - matchExpressions:
+                - key: master
+                  operator: DoesNotExist
+              - matchExpressions:
+                - key: dedicated
+                  operator: NotIn
+                  values:
+                  - skipper-ingress
+
+      hostPID: true
+      hostIPC: true
+      tolerations:
+        - key: karpenter.sh/disruption
+          operator: Exists
+        - key: node.kubernetes.io/not-ready
+          operator: Exists
+        - key: aws.amazon.com/spot
+          operator: Exists
+        - key: zalando.org/node-not-ready
+          operator: Exists
+          effect: NoSchedule
+        - key: nvidia.com/gpu
+          value: present
+          effect: NoSchedule
+
+      restartPolicy: Always
+
+      containers:
+      - name: wiz-sensor
+        image: wizio.azurecr.io/sensor:v1
+        imagePullPolicy: Always
+
+        securityContext:
+          capabilities:
+            add:
+            - SYS_ADMIN # moving between namespaces
+            - SYS_CHROOT # moving between namespaces
+            - SYS_RESOURCE # eBPF
+            - SYS_RAWIO # file hashing
+            - DAC_OVERRIDE # file hashing
+            - DAC_READ_SEARCH # file hashing
+            - NET_ADMIN # network events
+            - NET_RAW # network events
+            - IPC_LOCK # eBPF
+            - FOWNER # file hashing
+            - SYS_PTRACE # eBPF
+            - SYSLOG # kernel symbol resolve
+            - KILL # response
+          privileged: false
+          runAsNonRoot: true
+          runAsUser: 2202
+          runAsGroup: 2202
+          readOnlyRootFilesystem: true
+          seccompProfile:
+            type: Unconfined
+          seLinuxOptions:
+            user: "system_u"
+            role: "system_r"
+            level: "s0"
+            type: "spc_t"
+
+        env:
+        - name: MIN_SENSOR_COMPAT_BUILD_VER
+          value: "4142"
+        - name: HELM_CHART_COMPAT_VER
+          value: "1"
+        - name: WIZ_TMP_STORE
+          value: "/wiz-sensor-store/"
+        - name: WIZ_HOST_STORE
+          value: "/wiz-host-cache/"
+        - name: WIZ_RAMFS_STORE
+          value: "/tmp/"
+        - name: LOG_FILE
+          value: "/wiz-sensor-store/sensor.log"
```

Comment:
I see below the `tmp-store` `emptyDir` type volume that is backing the `/wiz-sensor-store` directory is only `100Mi` in size. This might not be suitable for logs as I see the log level set to `info`. We probably need a bigger volume here.

Also, please note that `emptyDir` volume by default is backed by the root disk of the node where this pod will run. (docs are [here](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir)). Sizing the volume to be too large might create issues for application nodes which will rely on the root disk capacity e-g to store the container image cache, among other things, and might have their own logs.

We might want to make a note of monitoring the log volume Wiz generates once we roll it out.

Reply:
Also, I see the `logging-destination` annotation missing on the pod templates in the application manifests, please add that. Example: https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/manifests/z-karpenter/deployment.yaml#L28

This will ensure that the Wiz logs are sent to Scalyr as well. This is a built-in feature provided in the infrastructure and the necessary information required to setup the connectivity with the Scalyr account is already injected to the pod when it's admitted, so you just need to provide this annotation and logs should show up in Scalyr.

That should make it safe to have a somewhat smaller volume for logging locally to the disk, since logs will be pushed to Scalyr already.